### PR TITLE
libcpuid: 0.5.1 -> 0.6.0

### PR DESCRIPTION
**Pull request by viraptor**
###### Description of changes

Update and drop unnecessary patches. (update required to bump cpu-x https://github.com/NixOS/nixpkgs/pull/195305)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by viraptor** [comment]
Result of `nixpkgs-review pr 195312` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.discover</li>
    <li>libsForQt514.discover</li>
  </ul>
</details>
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>crystal2nix</li>
  </ul>
</details>
<details>
  <summary>93 packages built:</summary>
  <ul>
    <li>appvm</li>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>cached-nix-shell</li>
    <li>cachix</li>
    <li>clickhouse</li>
    <li>colmena</li>
    <li>comma</li>
    <li>common-updater-scripts</li>
    <li>cpu-x</li>
    <li>crate2nix</li>
    <li>dep2nix</li>
    <li>dydisnix</li>
    <li>fusionInventory</li>
    <li>gnome.gnome-packagekit</li>
    <li>gnome.gnome-software</li>
    <li>simple-scan (gnome.simple-scan)</li>
    <li>go2nix</li>
    <li>haskellPackages.cabal2nix-unstable</li>
    <li>haskellPackages.cachix</li>
    <li>haskellPackages.hercules-ci-agent</li>
    <li>haskellPackages.hercules-ci-cli</li>
    <li>haskellPackages.hercules-ci-cnix-expr</li>
    <li>haskellPackages.hercules-ci-cnix-store</li>
    <li>haskellPackages.nix-paths</li>
    <li>haskellPackages.nix-serve-ng</li>
    <li>haskellPackages.nvfetcher</li>
    <li>haskellPackages.update-nix-fetchgit</li>
    <li>hci</li>
    <li>hercules-ci-agent</li>
    <li>hydra_unstable</li>
    <li>libcpuid</li>
    <li>libnixxml</li>
    <li>libsForQt5.discover (libsForQt515.discover ,plasma5Packages.discover)</li>
    <li>libsForQt5.packagekit-qt (libsForQt515.packagekit-qt ,plasma5Packages.packagekit-qt)</li>
    <li>libsForQt512.packagekit-qt</li>
    <li>libsForQt514.packagekit-qt</li>
    <li>libsForQt5_openssl_1_1.discover</li>
    <li>libsForQt5_openssl_1_1.packagekit-qt</li>
    <li>lispPackages.quicklisp-to-nix</li>
    <li>lispPackages.quicklisp-to-nix-system-info</li>
    <li>nil</li>
    <li>nix (nixVersions.nix_2_11 ,nixVersions.stable ,nixVersions.unstable)</li>
    <li>nix-bundle</li>
    <li>nix-direnv</li>
    <li>nix-doc</li>
    <li>nix-du</li>
    <li>nix-eval-jobs</li>
    <li>nix-index</li>
    <li>nix-pin</li>
    <li>nix-plugins</li>
    <li>nix-prefetch</li>
    <li>nix-prefetch-bzr</li>
    <li>nix-prefetch-cvs</li>
    <li>nix-prefetch-docker</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-hg</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-prefetch-svn</li>
    <li>nix-serve</li>
    <li>nix-simple-deploy</li>
    <li>nix-template</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nixStatic</li>
    <li>nixVersions.nix_2_10</li>
    <li>nixVersions.nix_2_7</li>
    <li>nixVersions.nix_2_8</li>
    <li>nixVersions.nix_2_9</li>
    <li>nixos-generators</li>
    <li>nixos-rebuild</li>
    <li>nixos-shell</li>
    <li>nixpkgs-review</li>
    <li>node2nix</li>
    <li>npins</li>
    <li>nuget-to-nix</li>
    <li>nvfetcher</li>
    <li>packagekit</li>
    <li>pantheon.appcenter</li>
    <li>pantheon.elementary-onboarding</li>
    <li>prefetch-yarn-deps</li>
    <li>python310Packages.nix-kernel</li>
    <li>python39Packages.nix-kernel</li>
    <li>rnix-lsp</li>
    <li>system-config-printer</li>
    <li>terranix</li>
    <li>tests.haskell.cabalSdist.hercules-ci-cnix-store</li>
    <li>tests.testers.nixosTest-example</li>
    <li>update-nix-fetchgit</li>
    <li>update-python-libraries</li>
    <li>vulnix</li>
    <li>wp4nix</li>
    <li>yarn2nix</li>
  </ul>
</details>

**Comment by viraptor** [comment]
ping @orivej @artuuge

File: pkgs/tools/misc/libcpuid/default.nix [hunk 1]
```diff
@@ -1,27 +1,16 @@
...
     owner = "anrieff";
     repo = "libcpuid";
     rev = "v${version}";
-    sha256 = "sha256-m10LdtwBk1Lx31AJ4HixEYaCkT7EHpF9+tOV1rSA6VU=";
+    sha256 = "sha256-XX+XFd1ehahIUGONAx2iOxetU0s6vONc7m91zEOWz5o=";
   };
 
-  patches = [
-    # Work around https://github.com/anrieff/libcpuid/pull/102.
-    ./stdint.patch
```
Comment (ajs124) [inline pkgs/tools/misc/libcpuid/default.nix:16]: The file should also be deleted

File: pkgs/tools/misc/libcpuid/default.nix [hunk 2]
```diff
@@ -1,27 +1,16 @@
...
-    # Fixes cross-compilation to NetBSD, remove me for libcpuid > 0.5.1
-    (fetchpatch {
-      name = "use-popcount-from-libc.patch";
-      url = "https://github.com/anrieff/libcpuid/commit/1acaf9980b55ae180cc08db218b9face28202519.patch";
-      sha256 = "0lvsv9baq0sria1f1ncn1b2783js29lfs5fv8milp54pg1wd5b7q";
-    })
-  ];
-
   nativeBuildInputs = [ autoreconfHook ];
```
Comment (ajs124) [inline pkgs/tools/misc/libcpuid/default.nix:14]: Is this still needed, if we're not patching anymore?

**Review by ajs124 — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/libcpuid/default.nix (hunk 2 above)
Comment (viraptor) [reply pkgs/tools/misc/libcpuid/default.nix:14]: Yes, it still handles the makefile generation / is required.

**Review by ajs124 — APPROVED** [review]

**Pull request merged**